jenkins/ci.suse.de/ansible-lint.yaml

```diff
@@ -0,0 +1,75 @@
+- job:
+    name: 'ansible-lint'
+    node: some-small-node
+    concurrent: true
+
+    logrotate:
+      numToKeep: 2000
+      daysToKeep: 300
+
+    properties:
+      - authorization:
+          cloud:
+            - job-build
+            - job-cancel
+            - job-configure
+            - job-delete
+            - job-discover
+            - job-read
+            - job-workspace
+            - run-delete
+            - run-update
+            - scm-tag
+          anonymous:
+            - job-read
+
+    parameters:
+      - string:
+          name: linter_venv
+          default: "$WORKSPACE/venv"
+          description: >-
+            The virtualenv to install dependencies into
+      - string:
+          name: git_url
+          description: >-
+            The URL to clone from
+      - string:
+          name: git_ref
+          description: >-
+            The git refspec to test
+      - string:
+          name: project_name
+          default: test-project
+          description: >-
+            The git refspec to test
+
+
+    builders:
+      - shell: |
+          set -x
+
+          if [ -d "${project_name}" ] ; then
+            pushd ${project_name}
+            git fetch ${git_url} ${git_ref}
+          else
+            git clone ${git_url} ${project_name}
+            pushd ${project_name}
+            git fetch ${git_url} ${git_ref}
+          fi
+          git reset --hard ${git_ref}
+          git clean -f
+          popd
+
+          if [ -r ${linter_env}/bin/activate ] ; then
+            source ${linter_env}/bin/activate
+          else
+            virtualenv ${linter_env}
+            source ${linter_env}/bin/activate
+          fi
+
+          pip install -U ansible ansible-lint
+
+          ansible-lint **/*.yml
```

Comment:
Note that this "lint" may report "spurious" failures in this scenario for Ardana ansible repos that have "cross" repo code references, since the Ardana code base when used in a real deployment is the amalgamated "whole" of all the ansible repos... This is why the existing Ardana ansible lint check takes a fair bit of time to run as it constructs the amalgamated code base to run the test against...

Reply:
Also by way of a warning some of the contortions associated with the existing Ardana ansible syntax check are related to issues with the verbosity of the output that that the ansible linter can generate, and stability issues seen with Jenkins/Java a few years back. Hopefully no longer an issue with the current versions of software being used.